src/api/web3.js

```diff
@@ -71,8 +71,17 @@ async function getWeb3() {
         networkState.expectedNetworkId
       )
 
-      // Checking if Web3 has been injected by the browser (Mist/MetaMask)
-      if (window.web3 && window.web3.currentProvider) {
+      if (window.ethereum) {
+        web3 = new Web3(window.ethereum)
+        try {
+          window.ethereum.enable().then(() => {
```

Comment:
This code won't work in the error condition - you need to `await` the promise instead of then `.then()`-ing it